### PR TITLE
Add DomainCreate method

### PR DESCRIPTION
**Pull request by polachok**

### CODE REVIEW

File: libvirt.go [hunk 1]
```diff
@@ -260,6 +260,29 @@ const (
 	StoragePoolsFlagZFS
 )
 
+// DomainCreateFlags specify options for starting domains
+type DomainCreateFlags uint32
+
+const (
+	// DomainCreateFlagNone is the default behavior.
+	DomainCreateFlagNone = 0
```
Comment (benlemasurier) [inline libvirt.go]: use `iota` here, e.g., https://github.com/digitalocean/go-libvirt/blob/master/libvirt.go#L116

File: libvirt.go [hunk 2]
```diff
@@ -343,6 +366,62 @@ func (l *Libvirt) Domains() ([]Domain, error) {
 	return result.Domains, nil
 }
 
+// DomainCreate starts specified domain
+func (l *Libvirt) DomainCreate(dom string) error {
```
Comment (benlemasurier) [inline libvirt.go]: no need for this duplicate function, `DomainCreateWithFlags` should be `DomainCreate(dom string, flags DomainCreateFlags)` where flags can be `0`.

**Comment by mdlayher** [comment]
Interesting, libvirt actually exposes two RPCs for this (one with flags and one without)?

**Comment by polachok** [comment]
>Interesting, libvirt actually exposes two RPCs for this (one with flags and one without)?

Yes. There're couple of others like DomainCreateWithXML, DomainCreateWithFIles

File: libvirt.go [hunk 3]
```diff
@@ -260,6 +260,26 @@ const (
 	StoragePoolsFlagZFS
 )
 
+// DomainCreateFlags specify options for starting domains
+type DomainCreateFlags uint32
+
+const (
```
Comment (benlemasurier) [inline libvirt.go]:
```go
const (
	// DomainCreateFlagPaused creates paused domain.
	DomainCreateFlagPaused = 1 << iota

	// DomainCreateFlagAutoDestroy destoy domain after libvirt connection closed.
	DomainCreateFlagAutoDestroy

	// DomainCreateFlagBypassCache avoid file system cache pollution.
	DomainCreateFlagBypassCache

	// DomainCreateFlagStartForceBoot boot, discarding any managed save
	DomainCreateFlagStartForceBoot

	// DomainCreateFlagStartValidate validate the XML document against schema
	DomainCreateFlagStartValidate
)
```

Comment (polachok) [inline libvirt.go]: ok, fixed

File: internal/constants/constants.go [hunk 4]
```diff
@@ -39,12 +39,14 @@ const (
 	ProcConnectOpen                = 1
 	ProcConnectClose               = 2
 	ProcConnectGetCapabilties      = 7
+	ProcDomainCreate               = 9
```
Comment (benlemasurier) [inline internal/constants/constants.go]: This can go away

File: libvirttest/libvirt.go [hunk 5]
```diff
@@ -418,6 +435,10 @@ func (m *MockLibvirt) handleRemote(procedure uint32, conn net.Conn) {
 		conn.Write(m.reply(testDestroyReply))
 	case constants.ProcDomainDefineXMLFlags:
 		conn.Write(m.reply(testDefineXML))
+	case constants.ProcDomainCreateWithFlags:
+		conn.Write(m.reply(testCreateWithFlags))
+	default:
+		fmt.Printf("unknown procedure %d", procedure)
```
Comment (benlemasurier) [inline libvirttest/libvirt.go]: `fmt.Fprintln(os.Stderr,...)`

Comment (polachok) [inline libvirttest/libvirt.go]: Shouldn't it be panic()?

Comment (benlemasurier) [inline libvirttest/libvirt.go]: `panic()` would work as well.

**Comment by benlemasurier** [comment]
Looks great! Feel free to make another commit adding yourself to `AUTHORS` and I'll get this merged in.

**Comment by benlemasurier** [comment]
thank you!

**Pull request merged**